### PR TITLE
Make spatial coordinates optional in RegionNDMap.interp_by_coord()

### PR DESCRIPTION
**Pull request by adonath**
See title. This pull request addresses #4072

### CODE REVIEW

**Review by AtreyeeS — CHANGES_REQUESTED** [review]
Thanks @adonath ! Maybe you can also fix `get_by_coord`?

**Comment by adonath** [comment]
Thanks @AtreyeeS! I'll include the change to `.get_by_coord()` in this PR...

File: gammapy/maps/region/ndmap.py [hunk 1]
```diff
@@ -395,8 +395,31 @@ def get_by_idx(self, idxs):
...
-        pix = self.geom.coord_to_pix(coords)
+        """Interpolate map values at the given map coordinates.
+
+        Parameters
+        ----------
+        coords : tuple, dict or `~gammapy.maps.MapCoord`
+            Coordinate arrays for each dimension of the map.  Tuple
+            should be ordered as (lon, lat, x_0, ..., x_n) where x_i
+            are coordinates for non-spatial dimensions of the map.
+            "lon" and "lat" are optional and will be take at the center
```
Comment (registerrier) [inline gammapy/maps/region/ndmap.py:406]: will be taken at

File: gammapy/maps/region/tests/test_ndmap.py [hunk 2]
```diff
@@ -211,6 +211,13 @@ def test_region_nd_map_get(region_map):
     values = region_map.get_by_coord((83.63, 21.51, energies[[0, -1]]))
     assert_allclose(values.squeeze(), [0, 5])
 
+    values = region_map.get_by_coord((energies[[0, -1]],))
+    assert_allclose(values.squeeze(), [0, 5])
+
+    values = region_map.get_by_coord({"energy": energies[[0, -1]]})
+    assert_allclose(values.squeeze(), [0, 5])
```
Comment (registerrier) [inline gammapy/maps/region/tests/test_ndmap.py:218]: Does it work with `region=None`? Maybe this should be tested as well?

File: gammapy/maps/region/tests/test_ndmap.py [hunk 3]
```diff
@@ -375,3 +382,32 @@ def test_region_nd_hdulist():
     assert hdulist[1].name == "SKYMAP"
     assert hdulist[2].name == "SKYMAP_BANDS"
     assert hdulist[3].name == "SKYMAP_REGION"
+
+
+def test_region_nd_map_interp_no_region():
```
Comment (registerrier) [inline gammapy/maps/region/tests/test_ndmap.py:387]: this answers the  previous comment...

File: gammapy/maps/region/geom.py [hunk 4]
```diff
@@ -513,6 +513,14 @@ def pix_to_idx(self, pix, clip=False):
         return tuple(idxs)
 
     def coord_to_pix(self, coords):
+        # inherited docstring
+
+        if isinstance(coords, dict):
+            coords.setdefault("skycoord", self.center_skydir)
```
Comment (registerrier) [inline gammapy/maps/region/geom.py:519]: what happens here if the `dict` contains already `lat` and `lon` keys?

**Review by registerrier — APPROVED** [review]
Thanks @adonath ! This looks good. I have left small inline questions/comments.

File: gammapy/maps/region/geom.py (hunk 4 above)
Comment (adonath) [reply gammapy/maps/region/geom.py:519]: Then `.setdefault()` does nothing...

Comment (registerrier) [reply gammapy/maps/region/geom.py:519]: It will add the `skycoord` to the list of keys and it will later be listed as one of the axes.

Comment (adonath) [reply gammapy/maps/region/geom.py:519]: Yes, of course! I forgot we treat `skycoord` separately from `lon` and `lat`. It was such a bad idea to support all of those, like `MapCoord`, `dict` (with `skycoord` and `lon` and `lat`) and `tuple`...it's so bad to maintain. We should get rid of it for Gammapy v2.0...

**Review by AtreyeeS — APPROVED** [review]
Thanks @adonath ! No further comments from my side

**Pull request merged**